docs/how-to/Bar-Memory.rst

```diff
@@ -0,0 +1,97 @@
+.. meta::
+   :description: Learn about BAR configuration in AMD GPUs and ways to troubleshoot physical addressing limit
+   :keywords: BAR memory, MMIO, GPU memory, Physical Addressing Limit, AMD, ROCm
+
+**************************************
+Troubleshoot BAR access limitation
+**************************************
+Direct Memory Access (DMA) to PCIe devices using Base Address Registers (BARs) can be restricted due to physical addressing limits. These restrictions can result in data access failures between the system components. Peer-to-peer (P2P) DMA is used to access resources such as registers and memory between devices. PCIe devices need memory-mapped input/output (MMIO) space for DMA, and these MMIO spaces are defined in the PCIe BARs.
+
+These BARs are a set of 32-bit or 64-bit registers that are used to define the memory space that PCIe devices can access. The CPU and other system devices also use these to access the resources of the PCIe devices. P2P DMA only works when one device can directly access the local BAR memory of another. If the BAR memory is more than the physical addressing limit of the device, the device will not be able to access the remote BAR. 
+
+To handle any BAR access issues that might occur, you need to be aware of the physical address limitations of the devices and understand the :ref:`BAR configuration of AMD GPUs <bar-configuration>`. This information is important when you need to set up additional MMIO apertures for PCIe devices in the system's physical address space.
+
+Handling physical address limitation
+=============================================
+When a system boots, the system BIOS allocates the physical address space for the components in the system, including system memory and MMIO apertures. There are two MMIO apertures: one allocated less than 4 GB of physical address space for 32-bit compatibility, and one more than 4 GB for devices needing more space.
```

Comment:
"There are two MMIO apertures: one allocated less than 4 GB of physical address space for 32-bit compatibility, and one more than 4 GB for devices needing more space."
I'd rephrase like so:
"On modern 64-bit platforms, there are generally two or more MMIO apertures: one located below 4 GB of physical address space for 32-bit compatibility, and one or more above 4 GB for devices needing more space."